File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int sum = 0;
        for(int num : nums) sum += num;
        int i = 0;
        int left_sum = 0;
        while(i < nums.size()) {
            sum -= nums[i];
            if(left_sum == sum) return i;
            else {
                left_sum += nums[i];
                i++;
            }
        }
        return -1;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int left_sum = 0, right_sum = 0;
        for(int num : nums) left_sum += num;
        int i = 0;
        while(i < nums.length) {
            left_sum -= nums[i];
            if(left_sum == right_sum) return i;
            else {
                right_sum += nums[i];
                i++;
            }
        }
        return -1;
    }
}